### ISSUE TITLE
The module name of mask decoder is not the same as the name of mask decoder in SAM2.

### ISSUE THREAD

**Issue by CloudyDory** [tianrun-chen/SAM-Adapter-PyTorch#102]
Hi, when running the SAM2-Adapter training code, I notice that the name of mask decoder in SAM2-Adapter is `self.mask_decoder`, but in the original SAM2 the module is named `self.sam_mask_decoder`. Therefore, when loading the SAM2 checkpoints in line 191 of `train.py`, the trained parameters of the mask decoder is not loaded, and is therefore trained from scratch in SAM2. I hope to know if this is the indented behavior of the code?

**Comment by tianrun-chen** [comment]
Kindly want to confirm which model of SAM2 are you using? We are using https://dl.fbaipublicfiles.com/segment_anything_2/092824/sam2.1_hiera_large.pt

**Comment by CloudyDory** [comment]
Hi, I am also using this file.

**Comment by liaozeyu20000110** [comment]
same question，I think the maskdecoder of sam2 was not loaded but trained from scratch
